modules/integration/tests-integration/tests-service/src/test/java/org/wso2/carbon/ml/lifecycle/test/Dataset9AbaloneTestCase.java

```diff
@@ -226,6 +232,31 @@ public void testBuildLinearRegressionModel() throws MLHttpClientException, IOExc
 
         // Model building should fail since Linear Regression cannot handle categorical response variables
         assertEquals("Model building did not fail as expected.", true, status);
+        testExportAsPMML();
+        testPublishAsPMML();
+    }
+
+    /**
+     * A test case for exporting a model in pmml format for failure model
+     *
+     * @throws MLHttpClientException
+     */
+    private void testExportAsPMML() throws MLHttpClientException {
+        response = mlHttpclient.exportAsPMML(modelId);
+        // Pmml download should fail
+        assertEquals("PMML download did not fail as expected", Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(),
+                response.getStatusLine().getStatusCode());
+    }
+
+    /**
+     * A test case for publishing a model to registry in pmml format for failure model
+     *
+     * @throws MLHttpClientException
+     */
+    private void testPublishAsPMML() throws MLHttpClientException {
+        response = mlHttpclient.doHttpPost("/api/models/" + modelId + "/publish?mode=pmml", null);
```

Comment:
As it appears, there's quite a few places in the code where doHttpPost() method is used in the form of mlHttpClient.doHttpPost(validParamValue, null).. Rather than passing null each and every time, maybe it's more cleaner to have an overloaded version of the same method taking only one valid parameter in thereby avoiding the need to pass null for the second parameter.